gprofiler/profilers/java.py

```diff
@@ -325,7 +333,12 @@ def _run_fdtransfer(self) -> None:
         """
         Start fdtransfer; it will fork & exit once ready, so we can continue with jattach.
         """
-        run_process([fdtransfer_path(), str(self.process.pid)], communicate=False)
+        run_process(
```

Comment:
OT: When running gprofiler as an executable, do we make sure not to leave processes like these running?

Reply:
Generally, gProfiler waits for all executed profilers (e.g py-spy) before it exits. Errors while waiting & bugs in the code are what this PR addresses (and possibly other processes like this one, which were invoked with no timeout at all).